### PR TITLE
Feat - Add integration and non-functional test documentation

### PR DESCRIPTION
**Pull request by jasondavindev**

### CODE REVIEW

File: tests/readme.md [hunk 1]
```diff
@@ -81,11 +79,43 @@ Is a type of software testing where individual units or components of a software
...
+- Scalability
+- Interoperability
+- Efficiency
+- Flexibility
+- Portability
+- Reusability
+
+## Software testing layers
+
+![Integration Testing](../images/integration-testing.png)
```
Comment (leon-lopes) [inline tests/readme.md:118]: Can we explain a bit more this image? I think we can add a few description.

File: tests/readme.md [hunk 2]
```diff
@@ -81,11 +79,43 @@ Is a type of software testing where individual units or components of a software
...
+Integration Testing is defined as a type of testing where software modules are integrated logically and tested as a group. A typical software project consists of multiple software modules, coded by different programmers. The purpose of this level of testing is to expose defects in the interaction between these software modules when they are integrated
+
+### Integration testing Roles
+
+- Mostly done by developers but testers can also participate, focusing on important or complex modules
 
 ### Non-function testing or performance testing
 
-TODO
+Non-Functional testing checks the Performance, reliability, scalability and other non-functional aspects of the software system. It is designed to test the readiness of a system as per nonfunctional parameters which are never addressed by functional testing.
```
Comment (toledompm) [inline tests/readme.md]:
```suggestion
Non-Functional testing checks the Performance, reliability, scalability and other non-functional aspects of the software system. It is designed to test the readiness of a system as per non-functional parameters which are never addressed by functional testing.
```

**Review by toledompm — APPROVED** [review]
:bike: some paragraphs have a full stop `.` while others dont :/

File: tests/readme.md (hunk 1 above)
Comment (jasondavindev) [reply tests/readme.md:118]: It's the compilation of all the above topics

**Review by M-Fernandes — APPROVED** [review]

**Pull request merged**